src/wp-includes/script-loader.php

```diff
@@ -2408,16 +2408,12 @@ function wp_enqueue_global_styles() {
 		return;
 	}
 
-	/*
+	/**
 	 * If we are loading CSS for each block separately, then we can load the theme.json CSS conditionally.
 	 * This removes the CSS from the global-styles stylesheet and adds it to the inline CSS for each block.
+	 * This filter has to be registered before we call wp_get_global_stylesheet();
 	 */
-	if ( $separate_assets ) {
-		add_filter( 'theme_json_get_style_nodes', 'wp_filter_out_block_nodes' );
-		// Add each block as an inline css.
-		wp_add_global_styles_for_blocks();
-	}
-
+	add_filter( 'gutenberg_get_style_nodes', 'filter_out_block_nodes' );
```

Comment:
We need to change the prefix here, e.g. to

```suggestion
	add_filter( 'wp_get_style_nodes', 'filter_out_block_nodes' );
```

Reply:
I'll apply this so I can approve the PR and mark it as "Needs Core commit".